packages/console/src/pages/Connectors/components/SetupModal.tsx

```diff
@@ -0,0 +1,113 @@
+import { ConnectorDTO, ConnectorType } from '@logto/schemas';
+import React, { useEffect, useMemo, useState } from 'react';
+import ReactModal from 'react-modal';
+import useSWR from 'swr';
+
+import Button from '@/components/Button';
+import Card from '@/components/Card';
+import CardTitle from '@/components/CardTitle';
+import IconButton from '@/components/IconButton';
+import ImagePlaceholder from '@/components/ImagePlaceholder';
+import RadioGroup, { Radio } from '@/components/RadioGroup';
+import UnnamedTrans from '@/components/UnnamedTrans';
+import { RequestError } from '@/hooks/use-api';
+import Close from '@/icons/Close';
+import * as modalStyles from '@/scss/modal.module.scss';
+
+import * as styles from './SetupModal.module.scss';
+
+type Props = {
+  isOpen: boolean;
+  onClose?: () => void;
+  type?: ConnectorType;
+};
+
+const SetupModal = ({ isOpen, onClose, type }: Props) => {
+  const [connectorId, setConnectorId] = useState<string>('');
+  const { data, error } = useSWR<ConnectorDTO[], RequestError>('/api/connectors');
+  const isLoading = !data && !error;
+
+  useEffect(() => {
+    if (isOpen) {
+      setConnectorId('');
+    }
+  }, [isOpen]);
+
+  const cardTitle = useMemo(() => {
+    if (type === ConnectorType.Email) {
+      return 'connectors.setup_title.email';
+    }
+
+    if (type === ConnectorType.SMS) {
+      return 'connectors.setup_title.sms';
+    }
+
+    return 'connectors.setup_title.social';
+  }, [type]);
```

Comment:
nit: as we have multiple places need to render different content based on different connector types, maybe we could have a static config some ware like. `DummyConnectorConfig[type].cardTitle`, `DummyConnectorConfig[type].details`

Reply:
Good call, but we only have `cardTitle` for now.